packages/accounts-base/url_client.js

```diff
@@ -1,11 +1,33 @@
-// By default, allow the autologin process to happen
-autoLoginEnabled = true;
+var Ap = AccountsClient.prototype;
 
 // All of the special hash URLs we support for accounts interactions
 var accountsPaths = ["reset-password", "verify-email", "enroll-account"];
 
+var savedHash = window.location.hash;
+window.location.hash = "";
```

Comment:
Saving the hash here means it no longer matters which `AccountsClient` instance calls `_attemptToMatchHash` first.